request_handlers.cc

```diff
@@ -12,88 +15,78 @@
 
 using boost::asio::ip::tcp;
 
+std::map<std::string, RequestHandler* (*)(void)>* request_handler_builders = nullptr;
+
+RequestHandler* RequestHandler::CreateByName(const char* type) {
+  const auto type_and_builder = request_handler_builders->find(type);
+  if (type_and_builder == request_handler_builders->end()) {
+    std::cout << "type_and_builder failed to find the type\n";
+    return nullptr;
+  }
+  return (*type_and_builder->second)();
+}
+
+// Request Handlers instantiated by lightning_server::run()
+// looks at child block of each route
+// constructs ServerConfig for each to look at root /exampleuri/something
+
+bool EchoRequestHandler::init(const std::string& uri_prefix,
+                              const NginxConfig& config) {
+  uri_prefix_ = uri_prefix;
+  config_ = ServerConfig(config);
+  // TODO: do logging/check return value of Build
+  config_.Build();
+  return true;
+}
+
 // Read in the request from the socket used to construct the handler, then
 // fill the response buffer with the necessary headers followed by the
 // original request.
-void EchoRequestHandler::handleRequest(const ServerConfig& server_config,
-                                       const char* request_buffer,
-                                       const size_t& request_buffer_size,
-                                       char* &response_buffer,
-                                       size_t& response_buffer_size) {
+RequestHandler::Status EchoRequestHandler::handleRequest(const Request& request,
+                                                         Response* response) {
+  std::cout << "EchoRequestHandler currently responding.\n";
 
   // Create response, defaulting to 200 OK status code for now
-  const std::string response_header = "HTTP/1.1 200 OK\r\nContent-Type: text/plain\r\n\r\n";
-  const size_t header_size = response_header.size();
-  response_buffer_size = header_size + request_buffer_size;
-
-  // Zero out memory using memset
-  // response_buffer size has a +1 for a null byte at the end
-  // std::memset((void *)&response_buffer, 0, sizeof(response_buffer));
-  response_buffer = new char[response_buffer_size + 1];
-  // std::memset((void *)&response_buffer, 0, sizeof(response_buffer));
-
-  // Copy in headers, the original request, and a terminating nullbyte
-  response_header.copy(response_buffer, header_size);
-  std::memcpy(&response_buffer[header_size], request_buffer, request_buffer_size);
-  std::memcpy(&response_buffer[header_size + request_buffer_size], "\0", 1);
-
-  std::cout << "~~~~~~~~~~Response~~~~~~~~~~\n" << response_buffer << std::endl;
+  // TODO: for some reason, browser is interpreting the response as a binary file
```

Comment:
Ah had actually fixed that, forgot to remove the comment. When i was changing some things, the buffer size was off by one